### ISSUE TITLE
[Bug] CPU usage shows unknown

### ISSUE THREAD

**Issue by 5a5cha** [Vaskivskyi/ha-asusrouter#491]
### The problem

After a fresh config and after 8h of runntime the integration shows "unknown" for the CPU usage.

Seen the morgning with "unknown" for Router an AP(Mesh-Node).
At time of the screenshoot only the Node seems to to be updated correct.

![Screenshot_20230125_093138_Home Assistant](https://user-images.githubusercontent.com/21234719/214516466-418eaf4c-3619-4096-a375-3d49888876db.jpg)

### Your device model

GT-AX11000

### Firmware type

Merlin

### Firmware version

388.1

### Integration version with the issue

0.18.1

### Method of the integration installation

HACS

### What version and type of Home Assistant installation do you use

2023.1.7

**Comment by Vaskivskyi** [comment]
Hey,

Is there anything in the HA log about the CPU sensor? Are any other sensors being `unknown` for the same/different amount of time without an obvious reason?

**Comment by 5a5cha** [comment]
See below:

2023-01-24 22:51:53.915 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration asusrouter which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-01-24 22:51:53.915 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-01-24 22:52:20.767 ERROR (MainThread) [homeassistant.helpers.device_registry] Unexpected exception from <function async_setup_cleanup.<locals>.cleanup at 0x7ff2ea5b9360>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/debounce.py", line 109, in _handle_timer_finish
    await task
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 808, in cleanup
    async_cleanup(hass, dev_reg, ent_reg)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 784, in async_cleanup
    dev_reg.async_remove_device(dev_id)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 521, in async_remove_device
    device = self.devices.pop(device_id)
  File "/usr/local/lib/python3.10/_collections_abc.py", line 963, in pop
    del self[key]
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 221, in __delitem__
    del self._identifiers[identifier]
KeyError: ('asusrouter', None)
2023-01-24 22:54:52.050 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entities for domain binary_sensor with platform asusrouter
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 438, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 589, in _async_add_entity
    device = device_registry.async_get_or_create(**processed_dev_info)  # type: ignore[arg-type]
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 353, in async_get_or_create
    device = self.async_update_device(
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 495, in async_update_device
    self.devices[device_id] = new
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 207, in __setitem__
    del self._identifiers[identifier]
KeyError: ('asusrouter', None)
2023-01-24 22:54:52.054 ERROR (MainThread) [homeassistant.components.button] Error adding entities for domain button with platform asusrouter
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 438, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 589, in _async_add_entity
    device = device_registry.async_get_or_create(**processed_dev_info)  # type: ignore[arg-type]
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 353, in async_get_or_create
    device = self.async_update_device(
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 495, in async_update_device
    self.devices[device_id] = new
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 207, in __setitem__
    del self._identifiers[identifier]
KeyError: ('asusrouter', None)
2023-01-24 22:54:52.067 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entities for domain binary_sensor with platform asusrouter
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 438, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 589, in _async_add_entity
    device = device_registry.async_get_or_create(**processed_dev_info)  # type: ignore[arg-type]
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 353, in async_get_or_create
    device = self.async_update_device(
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 495, in async_update_device
    self.devices[device_id] = new
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 207, in __setitem__
    del self._identifiers[identifier]
KeyError: ('asusrouter', None)

**Comment by Vaskivskyi** [comment]
The issue is fixed in release 0.18.2. Please, check it

**Comment by 5a5cha** [comment]
But there's another issue comming with it :-)
After update to 0.18.2 the integration won't start the binary_sensor and ha exit start with the following warning:
![image](https://user-images.githubusercontent.com/21234719/215291137-0f3c0096-0d15-41ab-a12c-bf34ccfff887.png)

Should a start a new bug?

**Comment by Vaskivskyi** [comment]
Sorry. Seems like there was a mistake in the update. Please check with 0.18.3

**Comment by drawgas** [comment]
@Vaskivskyi seems it is still happening even with 0.18.3
No errors in the log at least that I see.
CPU and WAN upload/download goes unavailable

<img width="999" alt="image" src="https://user-images.githubusercontent.com/3798139/215771655-4f2fee76-2a18-4c02-9bc1-0e251ada3a56.png">

**Comment by drawgas** [comment]
Sorry, it went unknown, my mistype :)
Just as you commented - it went unknown again (I've reloaded integration after I've posted initial comment)

<img width="309" alt="image" src="https://user-images.githubusercontent.com/3798139/215793309-6c368db8-ea5d-44c1-94dc-6ced8290da2e.png">

Router: RT-AX88U
Firmware: Merlin 388.1

> @drawgas, hi. Please, check, whether the entity become `unavailable` or `unknown`. Also, whether this happens to some other enties and whether they recover by themself or only after you manually reload the integration or restart HA.
> 
> Please, open a new issue with your findings
